### PR TITLE
[SPARK-43442][PS][CONNECT][TESTS] Split test module `pyspark_pandas_connect`

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Split test module `pyspark_pandas_connect`.

Add a new module `pyspark_pandas_slow_connect` which should keep in line with `pyspark_pandas_slow`

### Why are the changes needed?
`pyspark_pandas_connect` may take 3~4 hours

### Does this PR introduce _any_ user-facing change?
No, test-only

### How was this patch tested?
updated CI

### CODE REVIEW

File: dev/sparktestsupport/modules.py [hunk 1]
```diff
@@ -852,6 +852,22 @@ def __hash__(self):
...
         "pyspark.pandas.tests.connect.indexes.test_parity_base",
+    ],
+    excluded_python_implementations=[
+        "PyPy"  # Skip these tests under PyPy since they require numpy, pandas, and pyarrow and
+        # they aren't available there
+    ],
+)
+
+
+pyspark_pandas_slow_connect = Module(
```
Comment (zhengruifeng) [inline dev/sparktestsupport/modules.py:863]: the new module should be consistent with https://github.com/apache/spark/blob/master/dev/sparktestsupport/modules.py#L706-L733

**Comment by zhengruifeng** [comment]
It seems that `test_ops_on_diff_frames_*` is much slower in Connect

```
Starting test(python3.9): pyspark.pandas.tests.test_ops_on_diff_frames (temp output: /__w/spark/spark/python/target/e0c7d560-dd29-408d-9f7e-834cd2cc683a/python3.9__pyspark.pandas.tests.test_ops_on_diff_frames__j1jfu3kt.log)
Finished test(python3.9): pyspark.pandas.tests.test_ops_on_diff_frames (177s)
Starting test(python3.9): pyspark.pandas.tests.test_ops_on_diff_frames_groupby (temp output: /__w/spark/spark/python/target/bff4fd9b-cf0b-4847-bc19-08114b9b106c/python3.9__pyspark.pandas.tests.test_ops_on_diff_frames_groupby__0fb3fdq5.log)
Finished test(python3.9): pyspark.pandas.tests.test_ops_on_diff_frames_groupby (116s)
Starting test(python3.9): pyspark.pandas.tests.test_ops_on_diff_frames_slow (temp output: /__w/spark/spark/python/target/89a994de-9f04-4ed8-aa77-16d95b79956e/python3.9__pyspark.pandas.tests.test_ops_on_diff_frames_slow__zjjj9b5a.log)
Finished test(python3.9): pyspark.pandas.tests.test_ops_on_diff_frames_slow (163s)
```

```
Starting test(python3.9): pyspark.pandas.tests.connect.test_parity_ops_on_diff_frames (temp output: /__w/spark/spark/python/target/80844664-a5e0-4685-804e-8eff938c6681/python3.9__pyspark.pandas.tests.connect.test_parity_ops_on_diff_frames__mdlsorlh.log)
Finished test(python3.9): pyspark.pandas.tests.connect.test_parity_ops_on_diff_frames (1191s)
Starting test(python3.9): pyspark.pandas.tests.connect.test_parity_ops_on_diff_frames_groupby (temp output: /__w/spark/spark/python/target/bb5e908e-9185-46ac-826f-a2055f247f43/python3.9__pyspark.pandas.tests.connect.test_parity_ops_on_diff_frames_groupby__50hudnbv.log)
Finished test(python3.9): pyspark.pandas.tests.connect.test_parity_ops_on_diff_frames_groupby (443s) ... 8 tests were skipped
Starting test(python3.9): pyspark.pandas.tests.connect.test_parity_ops_on_diff_frames_slow (temp output: /__w/spark/spark/python/target/a20e748b-8b2d-493c-b926-ade588f0cb7a/python3.9__pyspark.pandas.tests.connect.test_parity_ops_on_diff_frames_slow__c78c3kwt.log)
Finished test(python3.9): pyspark.pandas.tests.connect.test_parity_ops_on_diff_frames_slow (911s) ... 7 tests were skipped
```

**Review by itholic — APPROVED** [review]

**Comment by xinrong-meng** [comment]
> It seems that `test_ops_on_diff_frames_*` is much slower atop Connect

Interesting.. May I ask if CI jobs run those Connect tests on a single node?

File: dev/sparktestsupport/modules.py (hunk 1 above)
Comment (dongjoon-hyun) [reply dev/sparktestsupport/modules.py:863]: Oh, I thought we add `_slow` as postfix.

So, the rational is to have the same test list with `pyspark_pandas_slow`?

If then, +1.

Comment (dongjoon-hyun) [reply dev/sparktestsupport/modules.py:863]: Please add some comments about the relationship.

**Review by HyukjinKwon — APPROVED** [review]

File: dev/sparktestsupport/modules.py (hunk 1 above)
Comment (zhengruifeng) [reply dev/sparktestsupport/modules.py:863]: yeah, let me add comments here

**Comment by zhengruifeng** [comment]
> > It seems that `test_ops_on_diff_frames_*` is much slower atop Connect
> 
> Interesting.. May I ask if CI jobs run those Connect tests on a single node?

yes, all tests are on single node. I am afraid there maybe some performance issue in pandas on spark on connect

**Comment by zhengruifeng** [comment]
In the [latest run](https://github.com/zhengruifeng/spark/actions/runs/4975618218/jobs/8902974576):

the `pyspark-pandas-connnect` took 1h 25m 21s, and `pyspark-pandas-slow-connnect` took 1h 45m 22s, so I think the split itself should be fine.

but still see :

```
Starting test(python3.9): pyspark.pandas.tests.test_ops_on_diff_frames (temp output: /__w/spark/spark/python/target/bcf91e64-da92-456c-a65f-2acbb5a57c5e/python3.9__pyspark.pandas.tests.test_ops_on_diff_frames__vhzam369.log)
Finished test(python3.9): pyspark.pandas.tests.test_ops_on_diff_frames (183s)
Starting test(python3.9): pyspark.pandas.tests.test_ops_on_diff_frames_groupby (temp output: /__w/spark/spark/python/target/21464771-1bb4-4e2e-8a76-cc5424bc3eee/python3.9__pyspark.pandas.tests.test_ops_on_diff_frames_groupby__3jskgr8q.log)
Finished test(python3.9): pyspark.pandas.tests.test_ops_on_diff_frames_groupby (121s)
Starting test(python3.9): pyspark.pandas.tests.test_ops_on_diff_frames_slow (temp output: /__w/spark/spark/python/target/4ec97d5c-ff3a-474f-aaa4-a3e08b503667/python3.9__pyspark.pandas.tests.test_ops_on_diff_frames_slow__bfsd6ob8.log)
Finished test(python3.9): pyspark.pandas.tests.test_ops_on_diff_frames_slow (167s)
```


```
Starting test(python3.9): pyspark.pandas.tests.connect.test_parity_ops_on_diff_frames (temp output: /__w/spark/spark/python/target/7d09fc80-b92b-4872-ae33-62f61aa4d2d9/python3.9__pyspark.pandas.tests.connect.test_parity_ops_on_diff_frames__3d49xqjs.log)
Finished test(python3.9): pyspark.pandas.tests.connect.test_parity_ops_on_diff_frames (1018s)
Starting test(python3.9): pyspark.pandas.tests.connect.test_parity_ops_on_diff_frames_groupby (temp output: /__w/spark/spark/python/target/6542f9a4-39fb-4605-9e7b-6d9a3e7b5a63/python3.9__pyspark.pandas.tests.connect.test_parity_ops_on_diff_frames_groupby__swoyoc14.log)
Finished test(python3.9): pyspark.pandas.tests.connect.test_parity_ops_on_diff_frames_groupby (350s) ... 8 tests were skipped
Starting test(python3.9): pyspark.pandas.tests.connect.test_parity_ops_on_diff_frames_slow (temp output: /__w/spark/spark/python/target/7880eb8c-9299-4e1c-af32-daa48aa10715/python3.9__pyspark.pandas.tests.connect.test_parity_ops_on_diff_frames_slow__y24aawsj.log)
Finished test(python3.9): pyspark.pandas.tests.connect.test_parity_ops_on_diff_frames_slow (732s) ... 7 tests were skipped
```

**Review by xinrong-meng — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by zhengruifeng** [comment]
thank you all for the reviews, @dongjoon-hyun @HyukjinKwon @xinrong-meng @itholic

**Pull request closed without merging**